server/sonar-server/src/main/java/org/sonar/server/computation/issue/SourceLinesCache.java

```diff
@@ -19,35 +19,36 @@
  */
 package org.sonar.server.computation.issue;
 
-import com.google.common.base.Function;
 import org.apache.commons.lang.StringUtils;
-import org.sonar.core.source.db.FileSourceDto;
-import org.sonar.server.db.DbClient;
-import org.sonar.server.source.db.FileSourceDb;
+import org.sonar.batch.protocol.output.BatchReport;
+import org.sonar.batch.protocol.output.BatchReportReader;
+import org.sonar.server.source.index.SourceLineDoc;
+import org.sonar.server.source.index.SourceLineIndex;
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 
-import java.io.InputStream;
-import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
- * Cache of the lines of the currently processed file. Only a single file
+ * Cache of the lines of the currently processed file. Only a <strong>single</strong> file
  * is kept in memory at a time. Moreover data is loaded on demand to avoid
  * useless db trips.
- * <p/>
+ * <br />
  * It assumes that db table FILE_SOURCES is up-to-date before using this
  * cache.
  */
```

Comment:
Please complete the comment to explain that data is first read from the report, then from the index